### PR TITLE
Migrate to `SharedWorker`

### PR DESCRIPTION
**Pull request by parfeon**
feat(shared-worker): migrate to shared worker

Use `SharedWorker` instead of `Service Worker` for better PubNub client instances feedback.

feat(shared-worker): add worker log verbosity flag

Add configuration option to enable debug log output from the subscription `SharedWorker`.

refactor: add declarations generation

Create types declaration files.

### CODE REVIEW

File: src/web/index.ts [hunk 1]
```diff
@@ -69,14 +69,17 @@ export default class PubNub extends PubNubCore<ArrayBuffer | string, PubNubFileP
...
+      // Inject subscription worker into transport provider stack.
+      transport = new SubscriptionWorkerMiddleware({
         clientIdentifier: clientConfiguration._instanceId,
         subscriptionKey: clientConfiguration.subscribeKey,
-        serviceWorkerUrl: configurationCopy.serviceWorkerUrl,
+        userId: clientConfiguration.getUserId(),
+        workerUrl: configurationCopy.subscriptionWorkerUrl,
         sdkVersion: clientConfiguration.getVersion(),
         logVerbosity: clientConfiguration.logVerbosity!,
+        workerLogVerbosity: platformConfiguration.subscriptionWorkerLogVerbosity!,
```
Comment (mohitpubnub) [inline src/web/index.ts:82]: I'm assuming that `platformConfiguration.subscriptionWorkerLogVerbosity` will always hold non-null value right?

File: src/transport/subscription-worker/subscription-worker.ts [hunk 2]
```diff
@@ -62,13 +84,29 @@ export type CancelRequestEvent = BasicEvent & {
...
 
-// region Service Worker
+// region Subscription Worker
+/**
+ * Shared subscription worker connected event.
+ *
+ * Event signal shared worker client that worker can be used.
+ */
+export type SharedWorkerConnected = {
+  type: 'shared-worker-connected';
```
Comment (mohitpubnub) [inline src/transport/subscription-worker/subscription-worker.ts:107]: :+1:

**Review by mohitpubnub — APPROVED** [review]
LGTM!

File: src/web/index.ts (hunk 1 above)
Comment (parfeon) [reply src/web/index.ts:82]: @mohitpubnub yup, there are defaults applied when final platform-specific configuration created [here](https://github.com/pubnub/javascript/blob/feat/use-shared-worker/src/web/components/configuration.ts#L120).

**Pull request merged**